HW-3-task1.js

```diff
@@ -0,0 +1,42 @@
+// 1.	There is a JSON file with a price list. All prices there are in dollars. You need to:
+// a.	convert all prices to euros (use a static exchange rate).
+// b.	add a currency field to the source and output objects
+// c.	save both objects into variables and log them into the console.
+
+const exchangeRate = 0.91;
+
+const fs = require('fs');
+
+const priceList = fs.readFileSync('task1-data.json', 'utf8');
+const newPriceList = JSON.parse(priceList);
+const arrayWithoutKeyData = newPriceList.data;
+
+for (let product in arrayWithoutKeyData) {
+  for (let data in arrayWithoutKeyData[product]) {
+    if (data === 'Prices') {
+      console.log(arrayWithoutKeyData[product][data]);
+    }
+  }
+}
+
+for (let product in arrayWithoutKeyData) {
+  product['Prices in euros'] = { a: 'b' };
+  for (let data in arrayWithoutKeyData[product]) {
+    if (data === 'Prices') {
+      for (let price in arrayWithoutKeyData[product][data]) {
+        arrayWithoutKeyData[product][data][price] *= exchangeRate;
+        arrayWithoutKeyData[product][data][price] += ' €';
+      }
+    }
+  }
+}
+
+for (let product in arrayWithoutKeyData) {
+  for (let data in arrayWithoutKeyData[product]) {
+    if (data === 'Prices') {
+      console.log(arrayWithoutKeyData[product][data]);
+    }
+  }
+}
```

Comment:
const fs = require('fs')
const path = require('path')
const fetch = require('node-fetch')

const key = '70770fd9697a791a27f89b1baff41c5b'
const exchengeUrl = `http://api.exchangeratesapi.io/v1/latest?access_key=${key}`

async function priceConvertor(sourceDataFile, baseCurrency, targetCurrency){

    const rawData = await fileReader(sourceDataFile)
    const exchangeRates = await getExchangeRate(baseCurrency, targetCurrency)
    const ratesDiff = exchangeRates['EUR'] - exchangeRates['USD'] // get the difference betwen rates
    const usdToEurRate = exchangeRates['EUR'] - ratesDiff

    // add currency field in USD price list
    const usdPriceList = JSON.parse(rawData)
    usdPriceList['data'].forEach(item => {
        item['Prices']['Currency'] = 'USD'
    });

    const eurPriceList = JSON.parse(rawData)
    eurPriceList['data'].forEach(item => {
        item['Prices']['Currency'] = 'EUR'

        item['Prices']['Price'] = (item['Prices']['Price'] * usdToEurRate).toFixed(2)
        item['Prices']['Retail Price'] = (item['Prices']['Retail Price'] * usdToEurRate).toFixed(2)
        item['Prices']['Wholesale Price'] = (item['Prices']['Wholesale Price'] * usdToEurRate).toFixed(2)
    });

    fs.writeFileSync('usd-price-list.json', JSON.stringify(usdPriceList), 'utf-8')
    fs.writeFileSync('eur-price-list.json', JSON.stringify(eurPriceList), 'utf-8')
}

async function fileReader(sourceDataFile) {
    const sourceDataPath = path.join(__dirname, sourceDataFile)
    return new Promise((resolve, reject) => {
        fs.readFile(sourceDataPath, 'utf-8', (err, data) => {
            if(err) {
                reject(err)
            } else {
                resolve(data)
            }
        })
    })
    .then(data => data)
    .catch(err => err)
}

async function getExchangeRate(baseCurrency, targetCurrency){
    const url = exchengeUrl
    return await fetch(url)
        .then(res => {
            return res.json()
        })
        .then(data => {
            return data.rates
        })
        .catch(err => err)
}


priceConvertor('data.json', 'USD', 'EUR')